components/arm/fake/fake.go

```diff
@@ -32,17 +42,50 @@ var fakeModelJSON []byte
 
 // AttrConfig is used for converting config attributes.
 type AttrConfig struct {
-	FailNew      bool   `json:"fail_new"`
-	FailValidate bool   `json:"fail_validate"`
-	ArmModel     string `json:"arm-model"`
+	ArmModel  string `json:"arm-model"`
+	ModelPath string `json:"model-path,omitempty"`
+}
+
+func modelFromName(model, name string) (referenceframe.Model, error) {
+	var custom referenceframe.Model
+	switch resource.ModelName(model) {
+	case xarm.ModelName6DOF.Name:
+		return xarm.Model(name, 6)
+	case xarm.ModelName7DOF.Name:
+		return xarm.Model(name, 7)
+	case ur.ModelName.Name:
+		return ur.Model(name)
+	case yahboom.ModelName.Name:
+		return yahboom.Model(name)
+	case eva.ModelName.Name:
+		return eva.Model(name)
+	case ModelName.Name:
+		return referenceframe.UnmarshalModelJSON(fakeModelJSON, name)
+	case "custom":
+		return custom, errors.New(customModelErr)
+	default:
+		return nil, errors.Errorf("fake arm cannot be created, unsupported arm_model: %s", model)
+	}
 }
 
 // Validate ensures all parts of the config are valid.
 func (config *AttrConfig) Validate(path string) error {
-	if config.FailValidate {
-		return errors.New("whoops! failed to validate")
+	var err error
+	if config.ArmModel == "" {
```

Comment:
Not sure this is quite what we want either.
Probably the logic flow should look something like this:
- if *both* ArmModel and ModelPath are populated, then that's an error.
- If one or the other is populated, then appropriately pull in the correct model or load the specified file
- if neither are populated, that's an error